pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -93,12 +94,12 @@ const (
 )
 
 // Status indicates the result of running a plugin. It consists of a code and a
-// message. When the status code is not `Success`, the status message should
+// message. When the status code is not `Success`, the status failures should
 // explain why.
 // NOTE: A nil Status is also considered as Success.
 type Status struct {
-	code    Code
-	message string
+	code     Code
+	failures []error
```

Comment:
I honestly think this is not proper and confusing. I don't really think we need it either, why not just concatenate the reasons in ```PredicateResultToFrameworkStatus``` into a single string that we store in message? This is basically what  ```utilerrors.NewAggregate``` returns.

Reply:
For each predicate/plugin, it may have multiple internal (atomic) failure reasons. Suppose the number is `n`, in theory, there could be `2^n` possible aggregated failure. Take `PodFitsResources` for example, there could be insufficient resource on cpu, memory, ephemeral storage, or extended resource, if we just concatenate the aggregated failures into a string, that string wouldn't be merged properly well with the failures in other nodes into the final histogram message:

For an aggregated string case:

- node1: \<insufficient cpu, memory, ephemeral stroage\>
- node2: \<insufficient cpu\>
- node3: \<insufficient memory\>

The message users can see will be:

```
0/3 nodes are available: 1 Insufficient cpu, memory, ephemeral stroage, 1 Insufficient cpu, 1 Insufficient memory
```

**NOTE**: the message length showed here becomes O(2<sup>n</sup>) instead of O(n)..

While with this PR, and also which is consistent with the old behavior:

- node1: \<insufficient cpu\>, \<insufficient memory\>, \<insufficient ephemeral stroage\>
- node2: \<insufficient cpu\>
- node3: \<insufficient memory\>

The message users can see will be:

```
0/3 nodes are available: 2 Insufficient cpu, 2 Insufficient memory, 1 Insufficient ephemeral stroage
```

Reply:
SG.